gcc/testsuite/rust/compile/issue-867.rs

```diff
@@ -0,0 +1,8 @@
+fn main() {
+    let _ = 42;
+    let a = _ + 123; // { dg-error "Use of '_' is not allowed on the right-side of an assignment" }
```

Comment:
This test is going to fail now :DD but no worries it's not a big deal. Once that's fixed I'll merge it :)

Reply:
> This test is going to fail now

became a bit lazy :D 
Fixed the testcase and this time made sure it passes. Also I rebased this PR too.